src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -57,11 +64,23 @@ public void locateComponentsForOnlineIntelligentScan() throws ComponentLocatorEx
         logger.info("Intelligent Scan mode does not support Component Location Analysis.");
         throw new ComponentLocatorException("Failed to generate Component Location Analysis file.");
     }
+    
+    private File serializeInputToJson(File saveInputFileDir, Input libInput) throws DetectUserFriendlyException {
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        String serializedLibInput = gson.toJson(libInput);
+        try {
+            File componentsSourceInputFile =  new File (saveInputFileDir, "components-source.json");
+            DetectFileUtils.writeToFile(componentsSourceInputFile, serializedLibInput);
+            return componentsSourceInputFile;
+        } catch (IOException ex) {
+            throw new DetectUserFriendlyException("Failed to create component location analysis output file", ex, ExitCodeType.FAILURE_UNKNOWN_ERROR);
+        }
+    }
 
-    private void runComponentLocator(List<Component> componentsList, File scanOutputFolder, File projectSrcDir) throws ComponentLocatorException {
+    private void runComponentLocator(List<Component> componentsList, File scanOutputFolder, File projectSrcDir) throws ComponentLocatorException, DetectUserFriendlyException {
         Input componentLocatorInput = generateComponentLocatorInput(componentsList, projectSrcDir);
         String outputFilepath = scanOutputFolder + "/" + DETECT_OUTPUT_FILE_NAME;
-
+        serializeInputToJson(scanOutputFolder, componentLocatorInput);
```

Comment:
Commenting here to make conversation easier. 

Shanty: "Is this something you want pushed to production? I use the serializeInputToJson(..) to aid in my local development but creating components-source.json and not cleaning it up is not something we want released with 8.11"

Madhu: "Yes, we should push this to production. The file is useful for diagnosis post GA.

Subsequent tasks are (not a priority):

Add the file to clean up tasks.
Include the file for diagnostics.
Fix component order to match between input and output."

@dmamidisynopsys This PR means both components-withlocations.json AND components-source.json will be in the Detect output. Can you confirm that we want components-source.json in the scan output subdirectory of Detect? If yes, we need to communicate this to QA and update our public documentation. We would also need to add the same error handling as we did for the output JSON (ie. "could not create input JSON" ... what the Detect log output looks like, handle other issues etc..) This is Detect UX and ideal I'd confirm this change with a PM, but given how we have worked with this feature so far -- I will ask Dheeraj.

Reply:
The source file should be added when -debug or -d option is set. That is the well defined pattern we use all over Detect. Not a new requirement, it is a technical pattern we should follow. @shantyk @gopannabd